### ISSUE TITLE
Spurious warning with nested "SELECT"

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#314]
Input:
```
SELECT
    user_id AS non_gdpr_users
FROM
    `benchmark_summary.dataset.audience_size_buckets` AS query
WHERE user_id NOT IN (SELECT user_id FROM `benchmark_summary.dataset.gdpr_safe_users`)
```

Output:
```
sqlfluff lint --dialect bigquery test.sql
Linting file datascience-benchmark-summary/test.sql
== [datascience-benchmark-summary/test.sql] FAIL
L:   5 | P:  43 | L026 | Reference '`benchmark_summary.dataset.gdpr_safe_users`'
                       | refers to table/view 'dataset' not found in the FROM
                       | clause.
L:   5 | P:  43 | L028 | Qualified reference
                       | '`benchmark_summary.dataset.gdpr_safe_users`' found in
                       | single table select which is inconsistent with previous
                       | references.
```

I don't think either of these warnings should be produced, as the nested `SELECT` is independent of the main query. Warnings like this may be appropriate if the nested `SELECT` is a [correlated subquery](https://en.wikipedia.org/wiki/Correlated_subquery) of the outer query.

**Comment by alanmcruickshank** [comment]
I think for now this is a bug. I'm going to put in a fix for this one. I'll see if I can sort the potential of correlated subqueries too.... 🤔

**Comment by alanmcruickshank** [comment]
Yep - fixed, including the correlated subquery case. PR on it's way.
